bvm/ballerina-profiler/src/main/java/io/ballerina/runtime/profiler/Main.java

```diff
@@ -0,0 +1,369 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.runtime.profiler;
+
+import io.ballerina.runtime.profiler.codegen.ProfilerClassLoader;
+import io.ballerina.runtime.profiler.codegen.ProfilerMethodWrapper;
+import io.ballerina.runtime.profiler.util.Constants;
+import io.ballerina.runtime.profiler.util.ProfilerException;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.jar.JarFile;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import static io.ballerina.runtime.profiler.ui.HttpServer.initializeHTMLExport;
+import static io.ballerina.runtime.profiler.ui.JsonParser.initializeCPUParser;
+import static io.ballerina.runtime.profiler.util.Constants.OUT;
+
+/**
+ * This class is used to as the driver class of the ballerina profiler.
+ *
+ * @since 2201.7.0
+ */
+public class Main {
+
+    static long profilerStartTime;
+    private static String balJarArgs = null;
+    static String balJarName = null;
+    static String targetDir = null;
+    static String skipFunctionString = null;
+    private static int balFunctionCount = 0;
+    static int moduleCount = 0;
+
+    static final List<String> INSTRUMENTED_PATHS = new ArrayList<>();
+    static final List<String> INSTRUMENTED_FILES = new ArrayList<>();
+    static final List<String> UTIL_INIT_PATHS = new ArrayList<>();
+    static final List<String> UTIL_PATHS = new ArrayList<>();
+
+    public static void main(String[] args) throws ProfilerException {
+        profilerStartTime = TimeUnit.MILLISECONDS.convert(System.nanoTime(), TimeUnit.NANOSECONDS);
+        addShutdownHookAndCleanup(); // Register a shutdown hook to handle graceful shutdown of the application
+        printHeader(); // Print the program header
+        handleProfilerArguments(args); // Handle command line arguments
+        extractProfiler(); // Extract the profiler used by the program
+        createTempJar(balJarName); // Create a temporary JAR file inside target/bin
+        initializeProfiling(balJarName); // Initialize profiling
```

Comment:
We do not need to pass `balJarName` since it is static and the methods are only called once. We can change the methods.